dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceFileHelper.java

```diff
@@ -29,12 +33,63 @@
 
 public final class SourceFileHelper {
 
+    private static final long BYTES_PER_MEGABYTE = 1024L * 1024L;
+    private static final long MAXIMUM_FILE_SIZE = BYTES_PER_MEGABYTE;
+    private static final long NOTEBOOK_MAXIMUM_FILE_SIZE = 3 * BYTES_PER_MEGABYTE;
     private static final Logger LOGGER = LoggerFactory.getLogger(SourceFileHelper.class);
 
     private SourceFileHelper() {
 
     }
 
+    public static void setContentWithLimits(SourceFile file, String content, String path) {
+        String limitedContent = content;
+        if (content != null) {
+            byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
```

Comment:
Following up on my earlier comment, does this mean we will have 3 copies of the blob? Well of any file, obviously, but huge files are the concern.

Guava's Utf8 class lets you get the length without duplicating the bytes, but then you still need the bytes for the test on line 55, so I don't know if it's worth it.